src/markets/pjm_market.jl

```diff
@@ -0,0 +1,117 @@
+        ),
+    ]
+end
+
+"""
+    get_timezone(market::PjmMarket) :: TimeZone
+
+Return the timezone of the given `market`.
+"""
+function get_timezone(market::PjmMarket)::TimeZone
+    return market.timezone
+end
+
+"""
+    get_day_ahead_hourly_lmp_raw_data(market::PjmMarket, start_date::ZonedDateTime, end_date::ZonedDateTime; folder::AbstractString=tempdir(), parser::Function=(args...) -> nothing)
+
+Download raw data for Day-Ahead Energy Market locational marginal pricing (LMP) data for all bus locations `market` and `start_date` to `end_date` and save it in `folder`.
+"""
+function get_day_ahead_hourly_lmp_raw_data(
+    market::PjmMarket,
+    start_date::ZonedDateTime,
+    end_date::ZonedDateTime;
+    folder::AbstractString = tempdir(),
+)::Nothing
+
+    # assert that the difference between the start and end date is less than 366 days
+    @assert Dates.value((end_date - start_date)) / 86400000 < 366 "Currently, we only support downloading data for less than 366 days."
+
+    directory = mkpath(joinpath(folder, market.directory))
+
+    start_date_str, start_hour_minute, end_date_str, end_hour_minute =
+        get_str_dates(start_date, end_date, market)
+    url = get_url_day_ahead_hourly_lmp(
+        start_date_str,
+        start_hour_minute,
+        end_date_str,
+        end_hour_minute,
+    )
+    access_key_dict = get_acess_key_headers()
+
+    file_path = joinpath(directory, "day_ahead_hourly_lmp.csv")
+    _download(url, file_path, access_key_dict)
+    return nothing
+end
+
+
+"""
+    get_day_ahead_hourly_lmp(market::PjmMarket, start_date::ZonedDateTime, end_date::ZonedDateTime; folder::AbstractString=tempdir(), parser::Function=(args...) -> nothing)
+
+Return a table with data for Day-Ahead Energy Market locational marginal pricing (LMP) data for all bus locations `market` and `start_date` to `end_date` and save it in `folder`.
+Parse the data using `parser` if required.
+"""
+function get_day_ahead_hourly_lmp(
+    market::PjmMarket,
+    start_date::ZonedDateTime,
+    end_date::ZonedDateTime;
+    parser::Bool = false,
+)
+    # assert that the difference between the start and end date is less than 366 days
+    @assert Dates.value((end_date - start_date)) / 86400000 < 366 "Currently, we only support downloading data for less than 366 days."
+
+    start_date_str, start_hour_minute, end_date_str, end_hour_minute =
+        get_str_dates(start_date, end_date, market)
+    url = get_url_day_ahead_hourly_lmp(
+        start_date_str,
+        start_hour_minute,
+        end_date_str,
+        end_hour_minute,
+    )
+    access_key_dict = get_acess_key_headers()
+
+    response = HTTP.get(url, headers = access_key_dict)
+
+    df = CSV.read(IOBuffer(response.body), DataFrame)
+    if parser
+        return parse_df_format(
+            df,
+            PJMDayAheadHourlyLMP_values_keys,
+            PJMDayAheadHourlyLMP_meta_keys,
+        )
+    else
+        return df
+    end
```

Comment:
If this function returns the df, you can call `parse_df_format` after. Therefore, ignoring type instability, I would make `parser` a function instead of a bool:

```suggestion
    parser::Function = (df) -> df,
)
    # assert that the difference between the start and end date is less than 366 days
    @assert Dates.value((end_date - start_date)) / 86400000 < 366 "Currently, we only support downloading data for less than 366 days."

    start_date_str, start_hour_minute, end_date_str, end_hour_minute =
        get_str_dates(start_date, end_date, market)
    url = get_url_day_ahead_hourly_lmp(
        start_date_str,
        start_hour_minute,
        end_date_str,
        end_hour_minute,
    )
    access_key_dict = get_acess_key_headers()

    response = HTTP.get(url, headers = access_key_dict)

    df = CSV.read(IOBuffer(response.body), DataFrame)
    return parser(df)
```

Reply:
And you can call this function like:

```
function get_day_ahead_hourly_lmp(
    market,
    start_date,
    end_date;
    (df) -> parse_df_format(
            df,
            PJMDayAheadHourlyLMP_values_keys,
            PJMDayAheadHourlyLMP_meta_keys,
        ),
)
```